### PR TITLE
Migrate HGLRCF722AIOV2

### LINKED ISSUE #632: Target HGLRCF722AIOV2 doesn't work for bf 4.5
### Describe the bug

Hi, i have new AIO FC: HGLRC SPECTER 25A AIO. It comes with BF 4.4.x.  
The problem is that it is impossible to update to BF 4.5 or newer versions. 
The BF configurator (10.10.0) that works great with other FC like betafpv, says CLOUD BUILD FAILED see log file.

The target is HGLRCF722AIOV2 
 
Please anyone had a solution? Please try to build it on your configurator. Mine log file is: 

https://build.betaflight.com/api/builds/5df8b09b0797dbb3e3d5b7ed0fd7e1ab/log

### To Reproduce

build with 10.10.0 conf.

### Expected behavior

\\

### Support ID

```txt
\\
```

### Flight controller

hglrc specter aio 25A

### Other components

_No response_

### How are the different components wired up (including port information)

_No response_

### Add any other context about the problem that you think might be relevant here

_No response_

### PR DESCRIPTION
**Pull request by haslinghuis**
- fixes #632
- please note sub optimal sharing SPI between OSD and blackbox
- missed migration in https://github.com/betaflight/unified-targets/pull/1175
- thanks to @ot0tot for improvements

### CODE REVIEW

**Comment by sghizz** [comment]
so, how can i add this config.h to official bf configs repo?

**Comment by haslinghuis** [comment]
There is no method besides building locally. Have to wait until merged.

**Comment by sghizz** [comment]
so, in the future they  add this target?

**Review by nerdCopter — APPROVED** [review]

File: configs/HGLRCF722AIOV2/config.h
Comment (nerdCopter) [inline configs/HGLRCF722AIOV2/config.h:122]:
```
./src/config/configs/HGLRCF722AIOV2/config.h:122:41: error: 'CW90' undeclared (first use in this function)
  122 | #define GYRO_1_ALIGN                    CW90
```

Comment (nerdCopter) [inline configs/HGLRCF722AIOV2/config.h:122]:
```suggestion
#define GYRO_1_ALIGN                    CW90_DEG
```

File: configs/HGLRCF722AIOV2/config.h
Comment (nerdCopter) [inline configs/HGLRCF722AIOV2/config.h:57]:
```suggestion
#define LED_STRIP_PIN                   PA8
#define UART1_TX_PIN                    PA9
#define UART2_TX_PIN                    PA2
```

File: configs/HGLRCF722AIOV2/config.h
Comment (nerdCopter) [inline configs/HGLRCF722AIOV2/config.h:61]:
```suggestion
#define UART6_TX_PIN                    PC6
```

**Review by nerdCopter — CHANGES_REQUESTED** [review]

**Review by nerdCopter — APPROVED** [review]

File: configs/HGLRCF722AIOV2/config.h
Comment (haslinghuis) [reply configs/HGLRCF722AIOV2/config.h]: Thanks. Long time ago I used this script

**Comment by nerdCopter** [comment]
@sghizz , try 
[betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip](https://github.com/user-attachments/files/18043129/betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip) while waiting for merge. I used the defines from your log.

File: configs/HGLRCF722AIOV2/config.h
Comment (ot0tot) [inline configs/HGLRCF722AIOV2/config.h:109]: F722 should use bitbang by default
```suggestion
#define DEFAULT_DSHOT_BITBANG         DSHOT_BITBANG_ON
```

File: configs/HGLRCF722AIOV2/config.h
Comment (ot0tot) [inline configs/HGLRCF722AIOV2/config.h:81]: I2C baro does not use CS pin
```suggestion

```

File: configs/HGLRCF722AIOV2/config.h
Comment (haslinghuis) [inline configs/HGLRCF722AIOV2/config.h:108]:
```suggestion
#define DEFAULT_DSHOT_BITBANG           DSHOT_BITBANG_ON
```

File: configs/HGLRCF722AIOV2/config.h
Comment (ot0tot) [inline configs/HGLRCF722AIOV2/config.h:102]:
```suggestion
    TIMER_PIN_MAP( 9, PA1,  2,  -1) \
    TIMER_PIN_MAP(10, PA0,  2,  -1)
```

**Comment by nerdCopter** [comment]
[betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip](https://github.com/user-attachments/files/18043232/betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip) @ `eff7f0e61e`

**Review by ot0tot — APPROVED** [review]

**Comment by nerdCopter** [comment]
when this merges, does @blckmn have to do anything, or will it show in the list automatically?

@sghizz : [betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip](https://github.com/user-attachments/files/18043529/betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip)

**Comment by sghizz** [comment]
> when this merges, does @blckmn have to do anything, or will it show in the list automatically?
> 
> @sghizz : [betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip](https://github.com/user-attachments/files/18043529/betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip)

great! it seems work. thank you! i had  this dump all from the original firmware 4.4.x of this FC and, it is diffent from your.. Can you check?

# resources
resource BEEPER 1 C13
resource MOTOR 1 B04
resource MOTOR 2 B05
resource MOTOR 3 B00
resource MOTOR 4 B01
resource MOTOR 5 A15
resource MOTOR 6 B03
resource MOTOR 7 B06
resource MOTOR 8 B07
resource SERIAL_TX 1 A09
resource SERIAL_TX 2 A02
resource SERIAL_TX 3 B10
resource SERIAL_TX 4 A00
resource SERIAL_TX 5 C12
resource SERIAL_TX 6 C06
resource SERIAL_TX 7 NONE
resource SERIAL_TX 8 NONE
resource SERIAL_TX 9 NONE
resource SERIAL_TX 10 NONE
resource SERIAL_RX 1 A10
resource SERIAL_RX 2 A03
resource SERIAL_RX 3 B11
resource SERIAL_RX 4 A01
resource SERIAL_RX 5 D02
resource SERIAL_RX 6 C07
resource SERIAL_RX 7 NONE
resource SERIAL_RX 8 NONE
resource SERIAL_RX 9 NONE
resource SERIAL_RX 10 NONE
resource SOFTSERIAL_TX 1 NONE
resource SOFTSERIAL_TX 2 NONE
resource SOFTSERIAL_RX 1 NONE
resource SOFTSERIAL_RX 2 NONE
resource I2C_SCL 1 B08
resource I2C_SCL 2 NONE
resource I2C_SCL 3 NONE
resource I2C_SCL 4 NONE
resource I2C_SDA 1 B09
resource I2C_SDA 2 NONE
resource I2C_SDA 3 NONE
resource I2C_SDA 4 NONE
resource LED 1 A13
resource LED 2 A14
resource LED 3 NONE
resource SPI_SCK 1 A05
resource SPI_SCK 2 B13
resource SPI_SCK 3 NONE
resource SPI_SCK 4 NONE
resource SPI_SDI 1 A06
resource SPI_SDI 2 B14
resource SPI_SDI 3 C11
resource SPI_SDI 4 NONE
resource SPI_SDO 1 A07
resource SPI_SDO 2 B15
resource SPI_SDO 3 C12
resource SPI_SDO 4 NONE
resource ESCSERIAL 1 NONE
resource ADC_BATT 1 C02
resource ADC_RSSI 1 C00
resource ADC_CURR 1 C01
resource ADC_EXT 1 NONE
resource BARO_CS 1 A04
resource BARO_EOC 1 NONE
resource BARO_XCLR 1 NONE
resource USB_MSC_PIN 1 NONE
resource FLASH_CS 1 C10
resource GYRO_EXTI 1 C04
resource GYRO_EXTI 2 NONE
resource GYRO_CS 1 B02
resource GYRO_CS 2 NONE
resource USB_DETECT 1 C14

# timer
timer B04 AF2
# pin B04: TIM3 CH1 (AF2)
timer B05 AF2
# pin B05: TIM3 CH2 (AF2)
timer B00 AF2
# pin B00: TIM3 CH3 (AF2)
timer B01 AF2
# pin B01: TIM3 CH4 (AF2)
timer A15 AF1
# pin A15: TIM2 CH1 (AF1)
timer B03 AF1
# pin B03: TIM2 CH2 (AF1)
timer B06 AF2
# pin B06: TIM4 CH1 (AF2)
timer B07 AF2
# pin B07: TIM4 CH2 (AF2)
timer A08 AF1
# pin A08: TIM1 CH1 (AF1)
timer A03 AF3
# pin A03: TIM9 CH2 (AF3)
timer A02 AF3
# pin A02: TIM9 CH1 (AF3)
timer A01 AF2
# pin A01: TIM5 CH2 (AF2)
timer A00 AF2
# pin A00: TIM5 CH1 (AF2)

# dma
dma SPI_SDO 1 NONE
dma SPI_SDO 2 NONE
dma SPI_SDO 3 NONE
dma SPI_SDO 4 NONE
dma SPI_SDI 1 NONE
dma SPI_SDI 2 NONE
dma SPI_SDI 3 NONE
dma SPI_SDI 4 NONE
dma SPI_TX 1 NONE
dma SPI_TX 2 NONE
dma SPI_TX 3 NONE
dma SPI_TX 4 NONE
dma SPI_RX 1 NONE
dma SPI_RX 2 NONE
dma SPI_RX 3 NONE
dma SPI_RX 4 NONE
dma ADC 1 1
# ADC 1: DMA2 Stream 4 Channel 0
dma ADC 2 NONE
dma ADC 3 NONE
dma UART_TX 1 NONE
dma UART_TX 2 NONE
dma UART_TX 3 NONE
dma UART_TX 4 NONE
dma UART_TX 5 NONE
dma UART_TX 6 NONE
dma UART_TX 7 NONE
dma UART_TX 8 NONE
dma UART_TX 9 NONE
dma UART_TX 10 NONE
dma UART_TX 11 NONE
dma UART_RX 1 NONE
dma UART_RX 2 NONE
dma UART_RX 3 NONE
dma UART_RX 4 NONE
dma UART_RX 5 NONE
dma UART_RX 6 NONE
dma UART_RX 7 NONE
dma UART_RX 8 NONE
dma UART_RX 9 NONE
dma UART_RX 10 NONE
dma UART_RX 11 NONE
dma pin B04 0
# pin B04: DMA1 Stream 4 Channel 5
dma pin B05 0
# pin B05: DMA1 Stream 5 Channel 5
dma pin B00 0
# pin B00: DMA1 Stream 7 Channel 5
dma pin B01 0
# pin B01: DMA1 Stream 2 Channel 5
dma pin A15 0
# pin A15: DMA1 Stream 5 Channel 3
dma pin B03 0
# pin B03: DMA1 Stream 6 Channel 3
dma pin B06 0
# pin B06: DMA1 Stream 0 Channel 2
dma pin B07 0
# pin B07: DMA1 Stream 3 Channel 2
dma pin A08 2
# pin A08: DMA2 Stream 3 Channel 6
dma pin A03 NONE
dma pin A02 NONE
dma pin A01 0
# pin A01: DMA1 Stream 4 Channel 6
dma pin A00 0
# pin A00: DMA1 Stream 2 Channel 6

# feature
feature -RX_PPM
feature -INFLIGHT_ACC_CAL
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -RANGEFINDER
feature -TELEMETRY
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DISPLAY
feature -OSD
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -RX_SPI
feature -ESC_SENSOR
feature -ANTI_GRAVITY
feature RX_SERIAL
feature GPS
feature OSD
feature AIRMODE
feature ANTI_GRAVITY

# serial
serial 20 1 115200 57600 0 115200
serial 0 131073 115200 57600 0 115200
serial 1 64 115200 57600 0 115200
serial 2 2 115200 57600 0 115200
serial 3 0 115200 57600 0 115200
serial 4 0 115200 57600 0 115200
serial 5 0 115200 57600 0 115200

# mixer
mixer QUADX

mmix reset

**Comment by sghizz** [comment]
> when this merges, does @blckmn have to do anything, or will it show in the list automatically?
> 
> @sghizz : [betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip](https://github.com/user-attachments/files/18043529/betaflight_4.5.2_STM32F7X2_HGLRCF722AIOV2_66e83511c.hex.zip)

PERFECT! Now cloud builds works!

**Comment by blckmn** [comment]
It should show in the list within a half day or so.

**Comment by sghizz** [comment]
> It should show in the list within a half day or so.

PERFECT! Now cloud build works! Thank you!!!

**Pull request merged**
